Updates for NekRS v25

nandu90 opened the pull request:
Work in progress

aprilnovak committed:
update include files in makefile

aprilnovak committed:
fixing errors

Commit:
update NekRS submodule to v25

Commit:
update nekRS submodule

Commit:
WIP

nandu90 committed:
updating NekInterface...WIP

nandu90 committed:
resolved NekInterface errors

nandu90 committed:
resolved NekRSProblem errors

nandu90 committed:
resolved NekTimeStepper errors

nandu90 committed:
resolved NekBoundaryFlux errors

nandu90 committed:
resolved NekPostprocessorValue errors

nandu90 committed:
resolved compilation errors

aprilnovak commented on src/base/NekRSProblem.C:
```diff
@@ -939,11 +941,13 @@ NekRSProblem::writeVolumeDisplacement(const int elem_id,
   auto vc = _nek_mesh->volumeCoupling();
   int id = vc.element[elem_id] * mesh->Np;
 
+  auto [x, y, z] = nekrs::host_xyz();
```
do we need `host_xyz()`, or could we get away with just the `host_x`, `host_y`, and `host_z` functions?

aprilnovak commented:
Thanks @nandu90 , this is a great start. I think the overall design will be compatible with the rest of Cardinal - can you point me to where the host-side solution fields are being updated from the device-side solutions? I may have missed it or it hasn't been added yet to the draft PR (in which case we'll need to add it probably in `NekRSProblem::syncSolutions` so that any Cardinal objects which try to access U, P, S have up to date values.

aprilnovak committed:
add data transfers

aprilnovak committed:
clang format

nandu90 committed:
bug fix

nandu90 committed:
update mesh and soln sync

nandu90 committed:
update nek_mesh/exact case

nandu90 committed:
update nek_mesh/first_order case

nandu90 committed:
update nek_mesh/second_order case

nandu90 committed:
update nek_mesh/sideset case

nandu90 committed:
WIP: fix for second order mesh

nandu90 committed:
fix for second order mesh

nandu90 committed:
one more fix for second order mesh

nandu90 committed:
bug fix for temperature

nandu90 committed:
update nek_temp/exact case

nandu90 committed:
update nek_temp cases

nandu90 committed:
fix transfer bugs

nandu90 committed:
update transfers/nek_temperature cases

nandu90 committed:
update transfers/nek_scalar_value cases

nandu90 committed:
update transfers/nek_source cases

nandu90 committed:
update transfers/nek_postprocessor cases

nandu90 committed:
update nek_errors/usrwrk_transfers cases

nandu90 committed:
update conduction/identical_interface/pyramid case

nandu90 committed:
update conduction/identical_interface/cube case

nandu90 committed:
update conduction/identical_volume/cube case

nandu90 committed:
update conduction/boundary_and_volume cases

nandu90 committed:
add sheet for tracking tests

nandu90 committed:
update case tracker

nandu90 committed:
Revise test cases for v25 with additional entries

Updated test cases for v25 to include new transfers and sources.

nandu90 committed:
minor typo

nandu90 committed:
Enhance trackv25.md with comment column

Added a comment column to the test cases for v25.

nandu90 committed:
Merge branch 'update-v25' of github.com:nandu90/cardinal into update-v25

nandu90 committed:
Update trackv25.md with new conduction tests

Added new test cases for conduction scenarios and updated statuses.

nandu90 committed:
Add nek_errors section to trackv25.md

kyleadm committed:
Fix transfers/nek_flux tests for nekRS v25 compatibility

- Migrate test input scripts from v23 to v25 format
- Replace options.getArgs("SCALAR00 DIFFUSIVITY", k) with
  options.getArgs("SCALAR00 DIFFUSIONCOEFF", k) in src/base/NekInterface.C
- Update expected max_error in gold/nek_out.csv to account for round-off
- Update test-tracking Markdown list

nandu90 committed:
Merge pull request #3 from kyleadm/update-v25-transfers-nek_flux

Fix transfers/nek_flux tests for nekRS v25 compatibility

nandu90 committed:
bug fix for rhocp

kyleadm committed:
Fix conduction/zero_flux tests for nekRS v25 compatibility

- Update test input scripts from v23 to v25 format
- Update test-tracking Markdown list

nandu90 committed:
Merge pull request #4 from kyleadm/update-v25-conduction-zero_flux

Fix conduction/zero_flux tests for nekRS v25 compatibility

kyleadm committed:
Fix conduction/nonidentical_interface/cylinders tests for nekRS v25 compatibility

- Update test input scripts from v23 to v25 format
- Update test-tracking Markdown list
- Note: two remaining TIMEOUT failures

kyleadm committed:
Fix conduction/nonidentical_volume/nondimensional tests for nekRS v25 compatibility
- Update test input scripts from v23 to v25 format
- Update test-tracking Markdown list
- Note: one remaining TIMEOUT failure

kyleadm committed:
Fix conduction/reverse_cht tests for nekRS v25 compatibility
- Update test input scripts from v23 to v25 format
- Update test-tracking Markdown list

nandu90 committed:
Merge pull request #6 from kyleadm/update-v25-conduction-reverse_cht

Fix conduction/reverse_cht tests for nekRS v25 compatibility

kyleadm committed:
Fix bug in NekInterface::get_temperature() and update test status.

nandu90 committed:
Merge pull request #5 from kyleadm/update-v25-conduction-nonidentical_interface-cylinders

Fix conduction/nonidentical_interface/cylinders tests for nekRS v25 compatibility

kyleadm committed:
Update test tracking: conduction/identical_volume/cube now passing

nandu90 committed:
Merge pull request #7 from kyleadm/update-v25-conduction-identical_volume-cube

Update test tracking: conduction/identical_volume/cube now passing

kyleadm committed:
Merge remote-tracking branch 'upstream/update-v25' into update-v25-conduction-nonidentical_volume-nondimensional

kyleadm committed:
Update test tracking: conduction/nonidentical_volume/nondimensional now passing

nandu90 committed:
Merge pull request #8 from kyleadm/update-v25-conduction-nonidentical_volume-nondimensional

Update v25 conduction/nonidentical_volume/nondimensional

kyleadm committed:
Update test input scripts from v23 to v25 format for conduction/nonidentical_volume/cylinder

kyleadm committed:
Update test tracking: conduction/nonidentical_volume/cylinder

kyleadm committed:
Fix o_usrwrk array sizing in NekInterface.C

Corrects an allocation bug that caused failures in
conduction/nonidentical_volume/cylinder. Both tests now pass
for this case, and the tracking list has been updated accordingly.

kyleadm committed:
Fix byte sizing for o_usrwrk

Keep o_usrwrk sized by elements and correct the copyFrom() call in
write_usrwrk_field_file() to use proper byte counts.

nandu90 committed:
Merge pull request #9 from kyleadm/update-v25-conduction-nonidentical_volume-cylinder

Update conduction/nonidentical_volume/cylinder tests for nekRS v25 compatibility

nandu90 committed:
Revise test results and comments in trackv25.md

Updated test results for various conduction and transfer tests, including comments on required updates.

nandu90 committed:
update to NekRS submodule to v26

nandu90 committed:
Merge branch 'update-v25' of github.com:nandu90/cardinal into update-v25

nandu90 committed:
fix temp copy problem

nandu90 committed:
fix some BC related routines

nandu90 committed:
fix par keys

nandu90 committed:
WIP

nandu90 committed:
remove redundant sync call

nandu90 committed:
update write_field functions

aprilnovak committed:
update pebble_1 tutorial

aprilnovak committed:
update nekrs standalone tutorial

aprilnovak committed:
update pebble_cht tutorial

aprilnovak committed:
update conduction only fhr tutorial.

aprilnovak committed:
update signature for UDF_Setup

aprilnovak committed:
Update VELOCITY block to FLUID VELOCITY

aprilnovak committed:
update PRESSURE to FLUID PRESSURE

aprilnovak committed:
update TEMPERATURE to SCALAR TEMPERATURE

aprilnovak committed:
writeInterval to checkpointInterval

aprilnovak committed:
replace writeControl with checkpointControl

aprilnovak committed:
update nek_spatial_bin_component_aux tests. Simplified tests and discovered that if temperature variable is missing we also need to protect against accessing scalar data objects not initialized.

aprilnovak committed:
update heat_transfer_coefficient test

aprilnovak committed:
update nek_output test

aprilnovak committed:
update multi_cht test

aprilnovak committed:
simplify exact mesh tests

aprilnovak committed:
update nek_temp directory

aprilnovak committed:
update transfers tests

aprilnovak committed:
update userobjects/gap test

aprilnovak committed:
update hexagonal_gap_layered tests

aprilnovak committed:
update interval tests

aprilnovak committed:
update layered_layered tests

aprilnovak committed:
update radial_layered tests

aprilnovak committed:
update subchannel_layered tests

aprilnovak committed:
update volume user objects

aprilnovak committed:
update sideset_layered tests

aprilnovak committed:
update invalid_field tests, which showed that the copy solution to host needs to protect against lack of scalars

aprilnovak committed:
update invalid_settings tests, which showed that we need to have isHeatFluxBoundary only check for udfNeumann since this is the intention of the error check

aprilnovak committed:
update invalid_transfer_pp test

aprilnovak committed:
update mpi_setup test

aprilnovak committed:
update no_occa_source_kernel

aprilnovak committed:
update no_temp_solve test

aprilnovak committed:
update no_temp_var test

aprilnovak committed:
update used_scratch test

aprilnovak committed:
update usrwrk_transfers test

aprilnovak committed:
update invalid_bid_postprocessor test

aprilnovak committed:
update dimensionless_numbers tests

aprilnovak committed:
update nek_heat_flux_integral tests

aprilnovak committed:
update nek_pressure_surface_force test

aprilnovak committed:
update nek_side_average

aprilnovak committed:
update nek_side_extrema test

aprilnovak committed:
update nek_side_integral test

aprilnovak committed:
update nek_weighted_side_average test

aprilnovak committed:
update nek_weighted_side_integral test

aprilnovak committed:
update zero_flux tests

aprilnovak committed:
update reverse_cht test

aprilnovak committed:
update nonidentical_interface test

aprilnovak committed:
update more tests

aprilnovak committed:
update adaptive_dt test

aprilnovak committed:
update sfr_pincell tests

aprilnovak committed:
update points test

aprilnovak committed:
update nek_point_value tests

aprilnovak committed:
update name of bc

aprilnovak committed:
correct nondimensional table

aprilnovak committed:
init volumetric data transfer

aprilnovak committed:
update nek_volume_integral tests

aprilnovak committed:
udate nek_volume_extrema tests

aprilnovak committed:
update nek_usrwrk_boundary_integral test

aprilnovak committed:
update nek_volume_average test

aprilnovak committed:
update tracking

aprilnovak committed:
Update sfr_7pin tutorial

aprilnovak committed:
update nek_viscous_surface_force test

aprilnovak committed:
try to update usrwrk test

aprilnovak committed:
update cht/pebble tests

aprilnovak committed:
remove unused files, these are not used in any test

nandu90 committed:
usrwrk write fix

aprilnovak committed:
documentation updates

aprilnovak committed:
update some nek_file_output tests, which showed that field files are now numbered starting from 0

aprilnovak committed:
update avg_plugin test

aprilnovak committed:
update avg_plugin test

aprilnovak committed:
Updating nek_stochastic tests

aprilnovak committed:
update NekRS submodule

aprilnovak committed:
update nek_ci submodule

aprilnovak committed:
update multiple_nek_apps tests

aprilnovak committed:
allow Cardinal to skip allocating usrwrk so that it wont complain if NekRS does, and also use new conventino for casename

aprilnovak committed:
update auxkernels

aprilnovak committed:
update NekInitAction header

aprilnovak committed:
delete ktauChannel test because it duplicates the ktauChannel test in nek_ci.

aprilnovak committed:
update from_restart test

aprilnovak committed:
update lowMach standalone test. Pressure diffs on order of 1%, which @nandu90 recommends to re-gold since it is within expected variation between code updates.

aprilnovak committed:
update two error tests

aprilnovak committed:
update start_time test

aprilnovak committed:
update fhr_reflector tutorial

aprilnovak committed:
trying to update gas_compact_cht case

aprilnovak committed:
update conj_ht test

aprilnovak committed:
update channel test

kyleadm committed:
Fix nek_stochastic/shift tests for nekRS v26 compatibility.

nandu90 committed:
Merge pull request #12 from kyleadm/update-v25-nek_stochastic-shift

Fix nek_stochastic/shift tests for nekRS v26 compatibility.

kyleadm committed:
WIP: add casename to prefix when writing field files.

When invoking checkpointWriter (an iofldFactory), the full prefix must now be passed explicitly, as the NekRS method no longer appears to append it automatically.
For example, when writing field files from Cardinal without including the case name in the prefix, the output files are named a00.nek500 instead of a00casename.nek500, which was the behavior in v23.
By adding the case name to the prefix passed to write_field_file(), the same naming format can be preserved in v26.

This is a WIP as there are still some issues with the filenames.

kyleadm committed:
Merge branch 'update-v25' into update-v25-nek_stochastic-driver_multi_fld

kyleadm committed:
WIP: Move the checkPointWriter() call outside of the isInitialized() conditional block.

If it is only called once during initialization, each subsequent MultiApp instance inherits the filename prefix from the first one.

For example, the expected output of the test:nek_stochastic.driver_multi_fld test is:

a00ethier0.f0000
a01ethier0.f0000
a02ethier0.f0000

However, when checkPointWriter() is called only once at initialization, the files are instead written as:

a00ethier0.f0000
a00ethier0.f0001
a00ethier0.f0002

This change ensures that the writer is configured separately for each MultiApp instance, restoring the original v23 behavior.

kyleadm committed:
Update the expected file names for the test:nek_stochastic.driver_multi_fld test.

In v23, the write-field method started field files at .f00001. The new approach starts numbering at .f00000.

kyleadm committed:
WIP: Fix tutorials/nek_stochastic input scripts for nekRS v26 compatibility.

kyleadm committed:
Update gold files for tutorials/nek_stochastic test cases

maxidalinger committed:
Erased lines that forced verbose in nekrs runs

kyleadm committed:
Update userobjects/side/nondimensional test input scripts for NekRS v26 compatibility.

- Update .par and .udf files for v26 changes
- Replace nrspre call (no longer built in v26) with nekrs --build-only
- Remove skip from the nondim test

The precompile test now passes, but the nondim test currently segfaults.

kyleadm committed:
Fix centroidFace loop bound (Np --> Nfp).

The loop iterated over mesh->Np (nodes per element) while indexing
face-node data using offsets based on mesh->Nfaces * mesh->Nfp.
This was observed to access invalid entries in vmapM and sgeo and lead to segfaults.

Restrict the loop to mesh->Nfp so it iterates only over nodes on the specified face.

nandu90 committed:
Merge pull request #14 from kyleadm/update-v25-tutorials-nek_stochastic

Update tutorials/nek_stochastic input scripts for nekRS v26 compatibility

nandu90 committed:
Merge pull request #16 from maxidalinger/update-v25

Erased lines that forced verbose in nekrs runs

nandu90 committed:
Merge pull request #17 from kyleadm/update-v25-userobjects-side-nondimensional

Update userobjects/side/nondimensional test input scripts for NekRS v26 compatibility + proposed bug fix

kyleadm committed:
Merge remote-tracking branch 'upstream/update-v25' into update-v25-tutorials-nek_stochastic-gold

kyleadm committed:
Update gold files for userobjects/side/nondimensional test cases.

kyleadm committed:
Replace references to the v23 MESH solver/field with the v26 GEOM solver/field.

kyleadm committed:
Update nek_errors/deformation test input scripts for NekRS v26 compatibility.

nandu90 committed:
Merge pull request #13 from kyleadm/update-v25-nek_stochastic-driver_multi_fld

Update source code to enable nek_stochastic/driver_multi_fld test to PASS

nandu90 committed:
Merge pull request #18 from kyleadm/update-v25-tutorials-nek_stochastic-gold

Update gold files for tutorials/nek_stochastic test cases

nandu90 committed:
Merge pull request #19 from kyleadm/update-v25-userobjects-side-nondimensional-gold

Update gold files for userobjects/side/nondimensional test cases.

nandu90 committed:
Merge pull request #20 from kyleadm/update-v25-nek_errors-deformation

Update nek_errors/deformation test input scripts for NekRS v26 compatibility + source code updates

nandu90 committed:
update msfr tutorial case

nandu90 committed:
Merge branch 'update-v25' of github.com:nandu90/cardinal into update-v25

nandu90 committed:
update msfr case - notes from MD
